### PR TITLE
Revert "Make build targets extendable (#2080)"

### PR DESCRIPTION
**Pull request by ViktorHofer**
This reverts commit e5b15175b9de059fbb38f9fc17280d93c66f0a8a.
Fixes https://github.com/dotnet/arcade/issues/2116

cc @markwilkie 

We don't need that undocumented extensibility point anymore in corefx.

### CODE REVIEW

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj [hunk 1]
```diff
@@ -38,6 +38,7 @@
     <_RepoRootOriginal>$(RepoRoot)</_RepoRootOriginal>
     <RepoRoot>$([System.IO.Path]::GetFullPath('$(RepoRoot)/'))</RepoRoot>
 
+    <_RemoveProps>Projects;Restore;QuietRestore;QuietRestoreBinaryLog;Build;Rebuild;Deploy;Test;IntegrationTest;PerformanceTest;Pack;Sign;Publish;NETCORE_ENGINEERING_TELEMETRY</_RemoveProps>
```
Comment (tmat) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj]: >QuietRestore;QuietRestoreBinaryLog [](start = 35, length = 34)

These are no longer used: `QuietRestore;QuietRestoreBinaryLog`. You can remove them from the list.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj [hunk 2]
```diff
@@ -101,10 +100,6 @@
       <_SolutionBuildTargets Include="PerformanceTest" Condition="'$(PerformanceTest)' == 'true'" />
     </ItemGroup>
 
-    <PropertyGroup>
-      <_RemoveProps>Projects;Restore;Deploy;Sign;Publish;NETCORE_ENGINEERING_TELEMETRY;@(_SolutionBuildTargets)</_RemoveProps>
-    </PropertyGroup>
```
Comment (tmat) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj]: Actually, let's keep this - it's cleaner with `@(_SolutionBuildTargets)`.

**Comment by tmat** [comment]
Thanks for cleaning this up!

**Comment by markwilkie** [comment]
Thanks @ViktorHofer !

**Pull request merged**